### PR TITLE
EZP-24617: Fixing toFieldDefinition method for ISBNConverter

### PR DESCRIPTION
**Pull request by tomaszmadeyski**
# Description

It fixes https://jira.ez.no/browse/EZP-24617

### More details:
Method resulted with true in both dataInt values: 1 and 0

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Legacy/Tests/Content/FieldValue/Converter/ISBNTest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<?php
+/**
```
Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/Tests/Content/FieldValue/Converter/ISBNTest.php]: Nitpick: AFAICS the most of the .php files in kernel have initial doc block separated by empty line, let's keep it that way.

**Comment by tomaszmadeyski** [comment]
@alongosz fixed in 28c3cca05294bcf918abe14bca5c5314e7f47401

File: eZ/Publish/Core/Persistence/Legacy/Tests/Content/FieldValue/Converter/ISBNTest.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        ]);
+
+        $this->converter->toFieldDefinition($storageDefinition, $fieldDef);
+
+        /** @var FieldSettings $fieldSettings */
+        $fieldSettings = $fieldDef->fieldTypeConstraints->fieldSettings;
+        self::assertSame($excpectedIsbn13Value, $fieldSettings['isISBN13']);
+    }
+
+    public function data_toFieldDefinition()
```
Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/Tests/Content/FieldValue/Converter/ISBNTest.php]: Sorry, didn't notice this before, PSR-2 method names must be in `camelCase`. 
I would use very explicit name such as `providerForTestToFieldDefinition` but might be shorter as long as there are no underscores.
Unfortunately php-cs-fixer won't protect us in these cases :(

**Comment by tomaszmadeyski** [comment]
@alongosz fixed in e2d7e85

**Comment by andrerom** [comment]
@tomaszmadeyski could you rebase for 6.7 branch?

**Comment by tomaszmadeyski** [comment]
@andrerom done

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ISBNConverter.php [hunk 3]
```diff
@@ -82,7 +82,7 @@ public function toFieldDefinition(StorageFieldDefinition $storageDef, FieldDefin
     {
         $fieldDef->fieldTypeConstraints->fieldSettings = new FieldSettings(
             array(
-                'isISBN13' => !empty($storageDef->dataInt1) ? (bool)$storageDef->dataInt1 : true,
+                'isISBN13' => !empty($storageDef->dataInt1),
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ISBNConverter.php]: So while this is correct, it still is a bit gotcha reading.

Would it make sense to make it more explicit so it's instantly what the code does here?

Like for instance: `$storageDef->dataInt1 !== null ? (bool)$storageDef->dataInt1 : true;`

Comment (tomaszmadeyski) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ISBNConverter.php]: heh, that's (almost) exactly how it was done before https://github.com/ezsystems/ezpublish-kernel/pull/1983/commits/f320360bd8e12f3b96b55d519ee2cd662430e8dd - I changed it according to @alongosz suggestions ;)

Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ISBNConverter.php]: Well, I guess it's very subjective. I don't like using ternary operator for result that already was determined by its condition. However I get the point about readability.

@tomaszmadeyski: @andrerom has the final say, so I'm sorry I made you change your original approach just to go back to it :) 
Please don't forget though that if `dataInt1` is `null`, it should return actually `false` (whole point of this PR) :-)

Comment (tomaszmadeyski) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ISBNConverter.php]: @andrerom @alongosz done in 52bb509

**Pull request merged**